xt_training/utils/sklearn_wrapper.py

```diff
@@ -0,0 +1,144 @@
+"""Wrappers for training/evaluating scikit learn models with xt-training.
+
+To use scikit learn models, a few simple modifications are required to the standard
+xt-training config format, as shown below:
+
+Usage:
+
+    Define train/val/test datasets as normal, where each sample returned should typically
+    be a row in a tabular dataset (this is generally the format suitable for sklearn).
+    Datasets should then each be wrapped in the SKDataset class:
+
+    >>> train_dataset = SKDataset(train_dataset)
+    >>> val_dataset = SKDataset(val_dataset)
+    >>> test_datasets = {k: SKDataset(v) for k, v in test_datasets.items()}
+
+    Similarly, sklearn models are made compatible by wrapping them with the SKInterface class:
+
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> model = RandomForestClassifier()
+    >>> model = SKInterface(model, output_dim=2, partial_fit=False)
+
+    Typically, sklearn models are fit to the entire dataset in one go, so we set batch_size and
+    epochs accordingly:
+
+    >>> epochs = 1
+    >>> batch_size = int(1e9)
+
+    Note that some sklearn models enable iterative fitting via the partial_fit method. In these
+    cases, set partial_fit=True when creating the SKInterface object, and you can also use more
+    than 1 epoch and a smaller batch_size.
+
+    Finally, since a pytorch optimizer is not required for training sklearn models, but it is
+    required by the xt-training `train` utility, a dummy optimizer is provided:
+
+    >>> optimizer = DummyOptimizer()
+
+    A scheduler should not be specified in the config.
+"""
+
+import torch
+from torch import nn
+from sklearn.utils.validation import check_is_fitted, NotFittedError
+
+
+class SKDataset:
+    """Dataset wrapper class to enable training scikit learn models with xt-training."""
+
+    def __init__(self, dataset):
+        self.dataset = dataset
+    
+    def __getitem__(self, i):
+        """Get a dataset sample and label
+
+        This returns (x, y), y to ensure that both x and y are passed to the sklearn model's
+        forward function, and are hence available for the fit() method.
+        """
+        x, y = self.dataset[i]
+        return (x, y), y
```

Comment:
why (x,y), y?

Reply:
Basically because we train using sklearn's fit method instead of a backward pass and optimizer step. Because of this, we have to pass both x and y to the forward method for the SKInterface module. Have a look at the SKInterface model class and it will hopefully make sense.